packages/material-ui-system/src/useThemeProps/useThemeProps.js

```diff
@@ -2,13 +2,7 @@ import getThemeProps from './getThemeProps';
 import useTheme from '../useTheme';
 
 export default function useThemeProps({ props, name, defaultTheme }) {
-  const contextTheme = useTheme(defaultTheme);
-  const more = getThemeProps({ theme: contextTheme, name, props });
-  const theme = more.theme || contextTheme;
-
-  return {
-    theme,
-    isRtl: theme.direction === 'rtl',
-    ...more,
-  };
+  const theme = useTheme(defaultTheme);
+  const mergedProps = getThemeProps({ theme, name, props });
+  return { ...mergedProps };
```

Comment:
No need to spend CPU cycles to clone

```suggestion
  return mergedProps;
```

Reply:
I think that there was complain that we are mutating props without it, but I will double check

Reply:
> I think that there was complain that we are mutating props without it, but I will double check

This warning could be hiding something else. AFAIk getThemeProps already clone it.

Reply:
agree to return `mergedProps` directly.